test/CMakeLists.txt

```diff
@@ -1,300 +1,320 @@
-project(test)
-
-include(CTest)
 add_subdirectory(googletest)
-include_directories(${gtest_SOURCE_DIR}/include ${gtest_SOURCE_DIR})
-include_directories(${gmock_SOURCE_DIR}/include ${gmock_SOURCE_DIR})
-include_directories(${PROJ_ROOT_INCLUDE_DIR})
+add_compile_definitions(NGEN_BMI_CPP_LIB_TESTS_ACTIVE)
 
-if (NGEN_ACTIVATE_PYTHON)
-    include_directories(${PROJECT_SOURCE_DIR}/extern/pybind11/include)
+# Natively support BMI C++ modules and pre-compile in the test_bmi_cpp mock/example.
+set(TEST_BMI_CPP_DIR ${NGEN_EXT_DIR}/test_bmi_cpp)
+git_update_submodule(${NGEN_EXT_DIR}/bmi-cxx)
```

Comment:
Should `bmi_cxx` be referenced directly from the top-level CMakeLists.txt?

Reply:
The thinking here was actually that we should stop using `include/bmi.hpp`--I think it is identical--and point to the upstream version...sort of as a matter of principle. Dogfooding, in a certain way. We are adamant that formulations must use an identical header to "ours" (for fairly obvious reasons of compatability)... what we really mean is that conformance to the official BMI header is required, both for semantic and technical reasons. So, do what we say and use a copy that can't be modified (practically)... and encourage the formulations to do the same, in fact.